settings.gradle

```diff
@@ -16,6 +16,39 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+plugins {
+  id 'com.gradle.enterprise' version '3.13.1'
+  id 'com.gradle.common-custom-user-data-gradle-plugin' version '1.10'
+}
+
+def isGithubActions = System.getenv('GITHUB_ACTIONS') != null
+def isJenkins = System.getenv('JENKINS_URL') != null
+def isCI = isGithubActions || isJenkins
+
+gradleEnterprise {
+  server = "https://ge.apache.org"
+  buildScan {
+    capture { taskInputFiles = true }
+    uploadInBackground = !isCI
+    publishAlways()
+    publishIfAuthenticated()
+    obfuscation {
+      // This obfuscates the IP addresses of the build machine in the build scan.
+      // Alternatively, the build scan will provide the hostname for troubleshooting host-specific issues.
+      ipAddresses { addresses -> addresses.collect { address -> "0.0.0.0"} }
```

Comment:
You can see here for an example of what all information is captured about infrastructure (as well as the build in general): https://ge.apache.org/s/ghik7lnrxcjkg#infrastructure

The impact here is that it will zero out the ip address, but will otherwise not affect the scan.